ocpi-toolkit-2.2.1/src/test/kotlin/com/izivia/ocpi/toolkit/tests/integration/mock/CustomLocationsCpoService.kt

```diff
@@ -0,0 +1,41 @@
+package com.izivia.ocpi.toolkit.tests.integration.mock
+
+import com.izivia.ocpi.toolkit.common.SearchResult
+import com.izivia.ocpi.toolkit.common.context.currentRequestMessageRoutingHeadersOrNull
+import com.izivia.ocpi.toolkit.common.context.currentResponseMessageRoutingHeaders
+import com.izivia.ocpi.toolkit.modules.locations.domain.Connector
+import com.izivia.ocpi.toolkit.modules.locations.domain.Evse
+import com.izivia.ocpi.toolkit.modules.locations.domain.Location
+import com.izivia.ocpi.toolkit.modules.locations.repositories.LocationsCpoRepository
+import java.time.Instant
+
+class CustomLocationsCpoService(
+    private val repository: LocationsCpoMongoRepository
+) : LocationsCpoRepository {
+    override suspend fun getLocations(
+        dateFrom: Instant?,
+        dateTo: Instant?,
+        offset: Int,
+        limit: Int?
+    ): SearchResult<Location> {
+        // This is an example, you will probably need something more specific
+        // if you want to handle message routing headers
+        val requestMessageRoutingHeaders = currentRequestMessageRoutingHeadersOrNull()
+        val responseMessageRoutingHeaders = currentResponseMessageRoutingHeaders()
```

Comment:
may I suggest adding a method to the RequestMessageRoutingHeaders that inverts itself? Something like 
```
invert(): ResponseMessageRoutingHeaders
```

alternative names could be `toResponseRoutingHeaders`, `toResponse`

Reply:
Yes I like the idea.
About the name, at firs glance t I liked `invert`, but it makes me feel it will return an inverted request routing headers. Therefore I would prefer `toResponse` despite it doesn't convey the inversion. 

Alternatively it could be a companion object method on the response, so that you would use it like this: `ResponseMessageRoutingHeaders.invertFromRequest(requestMessageRoutingHeaders)`

That's longer but I think it's more explicit.